File: tests/nested_func.py
def nested_func():
    def f1():
        print('f1')
    def f2():
        def f21():
            print('f2.1')
        print('f2')
        f21()
        f3()
    f1()
    f2()

def f3():
    print('f3')

File: tests/nested_func.m
function nested_func()

f1()
f2()
    function f1()
        disp('f1')
    end

    function f2()
        disp('f2')
        f21()
        f3()
        function f21()
            disp('f2.1')
        end
    end
end

function f3()
disp('f3');
end